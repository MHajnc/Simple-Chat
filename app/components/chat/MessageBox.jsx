import React from 'react';
import styles from '../../App.css';

export default class MessageBox extends React.Component {
  render() {
    if (this.props.currentuser == true) {
      return (
        <div className={styles.userMessage}>
          <div className={styles.userText}> {this.props.text} </div>
        </div>
      );
    }
    else {
      return (
        <div className={styles.incomingMessage}>
          <div className={styles.messageText}> {this.props.text} </div>
        </div>
      );
    }
  }
}
