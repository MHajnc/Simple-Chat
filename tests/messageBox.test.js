import React from 'react';
import renderer from 'react-test-renderer';

import MessageBox from '../app/components/chat/MessageBox';

test('Testing MessageBox Component -> currentuser="true"', () => {
  const out = renderer.create(
    <MessageBox currentuser={true} text="Hello!" />
  ).toJSON();

  expect(out).toMatchSnapshot();
});

test('Testing MessageBox Component -> currentuser="false"', () => {
  const out = renderer.create(
    <MessageBox currentuser={false} text="Hi man!" />
  ).toJSON();

  expect(out).toMatchSnapshot();
});