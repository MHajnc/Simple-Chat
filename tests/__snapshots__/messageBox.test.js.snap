// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MessageBox Component -> currentuser="false" 1`] = `
<div
  className="chat__messageList__incomingMessage"
>
  <div
    className="chat__messageList__messageText"
  >
     
    Hi man!
     
  </div>
</div>
`;

exports[`Testing MessageBox Component -> currentuser="true" 1`] = `
<div
  className="chat__messageList__userMessage"
>
  <div
    className="chat__messageList__userText"
  >
     
    Hello!
     
  </div>
</div>
`;
